syntax = "proto3";

package NAlice.NData;
option go_package = "a.yandex-team.ru/alice/protos/data/scenario/centaur/my_screen";
option java_package = "ru.yandex.alice.protos.data.scenario.centaur.my_screen";

import "mapreduce/yt/interface/protos/extension.proto";

message TCentaurWidgetData {
    option deprecated = true;
    option (NYT.default_field_flags) = SERIALIZATION_YT;
    option (NYT.default_oneof_flags) = VARIANT;
    
    message TVacantWidgetData {
    }
    message TMusicWidgetData {
    }
    message TNotificationWidgetData {
    }
    message TWeatherWidgetData {
    }
    message TNewsWidgetData {
    }
    message TTrafficWidgetData {
    }
    message TVideoCallWidgetData {
    }
    message TOpenWebviewWidgetData {
        option (NYT.default_field_flags) = SERIALIZATION_YT;
        option (NYT.default_oneof_flags) = VARIANT;
        oneof Source
        {
            option (NYT.variant_field_name) = "source";
            option (NYT.oneof_flags) = VARIANT;
            string Application = 1 [json_name = "application", (NYT.column_name) = "application"];
            string Url = 2 [json_name = "url", (NYT.column_name) = "url"];
        } 
    }
    bool Fixed = 1 [json_name = "fixed", (NYT.column_name) = "fixed"];
    oneof WidgetTypeData {
        option (NYT.variant_field_name) = "widget_type_data";
        option (NYT.oneof_flags) = VARIANT;
        TVacantWidgetData VacantWidgetData = 2 [json_name = "vacant_widget_data", (NYT.column_name) = "vacant_widget_data"];
        TMusicWidgetData MusicWidgetData = 3 [json_name = "music_widget_data", (NYT.column_name) = "music_widget_data"];
        TNotificationWidgetData NotificationWidgetData = 4 [json_name = "notification_widget_data", (NYT.column_name) = "notification_widget_data"];
        TWeatherWidgetData WeatherWidgetData = 5 [json_name = "weather_widget_data", (NYT.column_name) = "weather_widget_data"];
        TNewsWidgetData NewsWidgetData = 6 [json_name = "news_widget_data", (NYT.column_name) = "news_widget_data"];
        TOpenWebviewWidgetData OpenWebviewWidgetData = 7 [json_name = "open_webview_widget_data", (NYT.column_name) = "open_webview_widget_data"];
        TTrafficWidgetData TrafficWidgetData = 8 [json_name = "traffic_widget_data", (NYT.column_name) = "traffic_widget_data"];
        TVideoCallWidgetData VideoCallWidgetData = 9 [json_name = "video_call_widget_data", (NYT.column_name) = "video_call_widget_data"];
    }
}

message TCentaurWidgetConfigData {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    message TCustomWidgetTypeData {
    }
    string Id = 1 [json_name = "id", (NYT.column_name) = "id"];
    string WidgetType = 2 [json_name = "widget_type", (NYT.column_name) = "widget_type"];
    TCustomWidgetTypeData CustomWidgetTypeData = 3 [json_name = "custom_widget_type_data", (NYT.column_name) = "custom_widget_type_data"];
    bool Fixed = 4 [json_name = "fixed", (NYT.column_name) = "fixed"];
}

message TWidgetPosition {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    int32 Column = 1 [json_name = "column", (NYT.column_name) = "column"];
    int32 Row = 2 [json_name = "row", (NYT.column_name) = "row"];
}
